src/coreclr/CMakePresets.json

```diff
@@ -30,6 +30,21 @@
         }
       }
     },
+    {
+      "name": "osx-base",
+      "hidden": true,
+      "inherits": "base",
+      "cacheVariables": {
+        "CLR_CMAKE_TARGET_OS": "darwin"
```

Comment:
This is not correct. The variables are passed from build script. See https://github.com/dotnet/runtime/blob/e195258be55c6c3b930cdfedf906acf54896754b/src/coreclr/build-runtime.sh#L163

Reply:
Last time when I tested, `CLI_CMAKE_FALLBACK_OS` was required. It should be in .NET rid format (`osx`). Have you successfully perform a full build within IDE?

Reply:
Setting `osx` here not working, the error is below:

```
/Applications/CLion.app/Contents/bin/cmake/mac/aarch64/bin/cmake -DCMAKE_BUILD_TYPE=Debug --preset osx.arm64.Debug -S /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/src/coreclr -B /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/artifacts/obj/coreclr/osx.arm64.Debug
CMake Error at /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/eng/native/configureplatform.cmake:476 (message):
  Invalid host and target os/arch combination.  Host OS: darwin
Call Stack (most recent call first):
  /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/eng/native/configuretools.cmake:1 (include)
  /Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/eng/native/configurecompiler.cmake:5 (include)
  CMakeLists.txt:7 (include)

Preset CMake variables:

  CLR_CMAKE_HOST_ARCH="arm64"
  CLR_CMAKE_TARGET_OS="osx"
  CMAKE_INSTALL_PREFIX="/Users/gdc/Sandbox/dotnet-runtime-9.0.0-rc.2.24473.5/src/coreclr/../../artifacts/bin/coreclr/osx.arm64.Debug"


-- Configuring incomplete, errors occurred!

[Failed to reload]
```

But `darwin` does work.

Reply:
Yes `darwin` is correct: https://github.com/dotnet/runtime/blob/1434eeef6c9548c8be39cb0bb3aed11808146195/eng/native/configureplatform.cmake#L377

`CLI_CMAKE_FALLBACK_OS` is required for corehost:
https://github.com/dotnet/runtime/blob/1434eeef6c9548c8be39cb0bb3aed11808146195/src/native/corehost/CMakeLists.txt#L38-L40

and coreclr brings some components from corehost. Building in VS will also build the corresponding corehost subset, which requires `CLI_CMAKE_FALLBACK_OS` to be set. It should be `osx` and don't get confused with the `CLR_CMAKE_TARGET_OS` variable.